From 43969de910eb8feee728036415472bcf035bdc50 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Tue, 4 Jun 2019 10:21:20 +0000
Subject: [PATCH 1/1] efi_loader: avoid lost network packages

Prior to the patch booting an OrangePi PC via iPXE failed due to poor
network connection.

After receiving a packet immediately check if another packet is available.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_net.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/efi_loader/efi_net.c b/lib/efi_loader/efi_net.c
index d71c663068..3a598067db 100644
--- a/lib/efi_loader/efi_net.c
+++ b/lib/efi_loader/efi_net.c
@@ -517,6 +517,7 @@ static efi_status_t EFIAPI efi_net_receive
 	memcpy(buffer, net_rx_packet, net_rx_packet_len);
 	*buffer_size = net_rx_packet_len;
 	new_rx_packet = false;
+	efi_timer_check();
 out:
 	return EFI_EXIT(ret);
 }
-- 
2.20.1

